src/persist-client/src/read.rs

```diff
@@ -881,6 +858,53 @@ where
     }
 }
 
+impl<K, V, T, D> ReadHandle<K, V, T, D>
+where
+    K: Debug + Codec + Ord,
+    V: Debug + Codec + Ord,
+    T: Timestamp + Lattice + Codec64,
+    D: Semigroup + Codec64 + Send + Sync,
+{
+    /// Generates a [Self::snapshot], and fetches all of the batches it
+    /// contains.
+    ///
+    /// To keep memory usage down when reading a snapshot that consolidates
+    /// well, this consolidates as it goes.
```

Comment:
I think either add a note that the final result may yet _not_ be consolidated—or slap a consolidate on the end?

Reply:
> Also curious if you think we should put a consolidate at the end and then document that the output is guaranteed to be consolidated

Missed this in your PR description, oops. 🤦🏽‍♂️ If we decide not to consolidate, I do think it's worth making the docstring mention that we're not consolidating a bit more explicitly.